### PR TITLE
BUG: Fix to_datetime(errors='coerce') not swallowing all parser exceptions...

### LINKED ISSUE #28299: to_datetime(foo, errors='coerce') does not swallow all errors
#### Code Sample

```python
# this fails with ValueError in 0.25.1:
pandas.to_datetime('200622-12-31', errors='coerce')
# but returns Timestamp('2022-06-21 19:00:00') in pandas 0.23.4

# this also fails: 
pandas.to_datetime('111111-24-11', errors='coerce')

# but this does not:
pandas.to_datetime('111111-23-11', errors='coerce')
```
#### Problem description
I have some text files with malformed dates, which at one point I will process with the above code. While trying to migrate my code from 23.4 to 25.1 I got the following:

```python
.../my_file.py in <module>
----> 1 pandas.to_datetime('200622-12-31', errors='coerce')

.../lib/python3.7/site-packages/pandas/util/_decorators.py in wrapper(*args, **kwargs)
    206                 else:
    207                     kwargs[new_arg_name] = new_arg_value
--> 208             return func(*args, **kwargs)
    209 
    210         return wrapper

.../lib/python3.7/site-packages/pandas/core/tools/datetimes.py in to_datetime(arg, errors, dayfirst, yearfirst, utc, box, format, exact, unit, infer_datetime_format, origin, cache)
    794             result = convert_listlike(arg, box, format)
    795     else:
--> 796         result = convert_listlike(np.array([arg]), box, format)[0]
    797 
    798     return result

.../lib/python3.7/site-packages/pandas/core/tools/datetimes.py in _convert_listlike_datetimes(arg, box, format, name, tz, unit, errors, infer_datetime_format, dayfirst, yearfirst, exact)
    461             errors=errors,
    462             require_iso8601=require_iso8601,
--> 463             allow_object=True,
    464         )
    465 

.../lib/python3.7/site-packages/pandas/core/arrays/datetimes.py in objects_to_datetime64ns(data, dayfirst, yearfirst, utc, errors, require_iso8601, allow_object)
   1982             return values.view("i8"), tz_parsed
   1983         except (ValueError, TypeError):
-> 1984             raise e
   1985 
   1986     if tz_parsed is not None:

.../lib/python3.7/site-packages/pandas/core/arrays/datetimes.py in objects_to_datetime64ns(data, dayfirst, yearfirst, utc, errors, require_iso8601, allow_object)
   1973             dayfirst=dayfirst,
   1974             yearfirst=yearfirst,
-> 1975             require_iso8601=require_iso8601,
   1976         )
   1977     except ValueError as e:

pandas/_libs/tslib.pyx in pandas._libs.tslib.array_to_datetime()

pandas/_libs/tslib.pyx in pandas._libs.tslib.array_to_datetime()

ValueError: offset must be a timedelta strictly between -timedelta(hours=24) and timedelta(hours=24).
```

#### Expected Output
The main expectation is that an exception is not raised.

I would probably expect ` pandas.to_datetime('200622-12-31', errors='coerce')` to return NaT, but pandas 23.4 seems to parse it into `Timestamp('2022-06-21 19:00:00')`

#### Output of ``pd.show_versions()``

<details>                                                                                                                                                                                                                                          

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.3.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-58-generic
machine          : x86_64
processor        : 
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 0.25.1
numpy            : 1.16.4
pytz             : 2019.1
dateutil         : 2.8.0
pip              : 19.1.1
setuptools       : 41.0.1
Cython           : 0.29.10
pytest           : 4.6.3
hypothesis       : 4.7.3
sphinx           : None
blosc            : None
feather          : 0.4.0
xlsxwriter       : 1.1.8
lxml.etree       : 4.3.3
html5lib         : 1.0.1
pymysql          : 0.9.3
psycopg2         : 2.7.7 (dt dec pq3 ext lo64)
jinja2           : 2.10.1
IPython          : 7.5.0
pandas_datareader: None
bs4              : 4.7.1
bottleneck       : 1.2.1
fastparquet      : None
gcsfs            : 0.3.0
lxml.etree       : 4.3.3
matplotlib       : 3.1.1
numexpr          : 2.6.9
odfpy            : None
openpyxl         : 2.6.2
pandas_gbq       : None
pyarrow          : 0.13.0
pytables         : None
s3fs             : None
scipy            : 1.3.0
sqlalchemy       : 1.2.14
tables           : 3.5.2
xarray           : None
xlrd             : 1.2.0
xlwt             : 1.3.0
xlsxwriter       : 1.1.8


</details>

### PR DESCRIPTION
**Pull request by miggec**
… (#28299)

- [x] closes #28299
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.25.2.rst [hunk 1]
```diff
@@ -19,7 +19,7 @@ Categorical
 Datetimelike
 ^^^^^^^^^^^^
 
--
+- Bug in :func:`to_datetime` where passing arrays of malformed ``str`` while attempting to coerce errors could incorrectly lead to raising errors instead (:issue:`28299`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.2.rst]: move to 1.0

**Review by jreback — CHANGES_REQUESTED** [review]
cc @jbrockmendel

**Comment by jreback** [comment]
this *might* solve https://github.com/pandas-dev/pandas/issues/25143, https://github.com/pandas-dev/pandas/issues/23447, https://github.com/pandas-dev/pandas/issues/9107

if you'd have a look (if so pls add a test for whatever this solves)

File: doc/source/whatsnew/v0.25.2.rst (hunk 1 above)
Comment (squigmig) [reply doc/source/whatsnew/v0.25.2.rst]: thanks, done

**Comment by squigmig** [comment]
> this _might_ solve #25143, #23447, #9107
> 
> if you'd have a look (if so pls add a test for whatever this solves)

Looks like none of these is affected. At least one seems to be handled by a different change

**Review by TomAugspurger — APPROVED** [review]

**Comment by jreback** [comment]
@squigmig 

> Looks like none of these is affected. At least one seems to be handled by a different change

care to do a PR with a test for (which?) one

**Comment by squigmig** [comment]
> > Looks like none of these is affected. At least one seems to be handled by a different change
> 
> care to do a PR with a test for (which?) one

sure, have raised #28386

...

Failing checks on this PR seem to be intermittent. What is the preferred way to resolve or force a re-run?

https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=17284
Checks: Setup environment and build pandas is probably the originator of the rest of the failures
```conda env create -q --file=environment.yml
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done

CondaHTTPError: HTTP 404 NOT FOUND for url <https://repo.anaconda.com/pkgs/main/linux-64/widgetsnbextension-3.5.1-py37_0.conda>
Elapsed: 00:00.006773
CF-RAY: 5148a6736939e0b2-IAD

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.

##[error]Bash exited with code '1'.
##[section]Finishing: Setup environment and build pandas```

**Comment by TomAugspurger** [comment]
Pushing an empty commit, or asking a maintainer. I've restarted the build.

File: doc/source/whatsnew/v1.0.0.rst [hunk 2]
```diff
@@ -115,10 +115,10 @@ Datetimelike
 - Bug in :meth:`Series.__setitem__` incorrectly casting ``np.timedelta64("NaT")`` to ``np.datetime64("NaT")`` when inserting into a :class:`Series` with datetime64 dtype (:issue:`27311`)
 - Bug in :meth:`Series.dt` property lookups when the underlying data is read-only (:issue:`27529`)
 - Bug in ``HDFStore.__getitem__`` incorrectly reading tz attribute created in Python 2 (:issue:`26443`)
+- Bug in :func:`to_datetime` where passing arrays of malformed ``str`` while attempting to coerce errors could incorrectly lead to raising errors instead (:issue:`28299`)
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v1.0.0.rst]: 'while attempting to coerce errors' --> 'with `errors="coerce"`'

'could incorrectly lead to raising errors instead' --> 'could incorrectly raise [???]`

File: pandas/_libs/tslib.pyx [hunk 3]
```diff
@@ -609,16 +609,17 @@ cpdef array_to_datetime(ndarray[object] values, str errors='raise',
                             py_dt = parse_datetime_string(val,
                                                           dayfirst=dayfirst,
                                                           yearfirst=yearfirst)
+                            # If the dateutil parser returned tzinfo, capture it
+                            # to check if all arguments have the same tzinfo
+                            tz = py_dt.utcoffset()
```
Comment (jbrockmendel) [inline pandas/_libs/tslib.pyx]: can the utcoffset() call raise?  if not, why move this up?

**Review by jreback — APPROVED** [review]

**Pull request merged**